hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java

```diff
@@ -168,4 +171,28 @@ public String retryQuestion(String questionId, String comments, Question questio
                 .build());
         return askQuestion(question);
     }
+
+    public BinaryResource getAvatar(String agentId)
+    {
+        try
+        {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(String.format(config.getAvatarUrl(), agentId)))
+                    .headers(authService.getAuthHeaders())
+                    .GET()
+                    .build();
+
+            HttpResponse<InputStream> httpResponse = client.send(request, BodyHandlers.ofInputStream());
+
+            ensureCorrectHttpStatusReturned(SC_OK, httpResponse);
+            log.atDebug().log("Agent with id {} received avatar successfully", agentId);
+
+            File tempImageFile = TempFileProvider.createTempFile(httpResponse.body(), "RenditionsApi-", "png");
+            return new FileBinaryResource(tempImageFile, null);
```

Comment:
We can use a constructor with a single argument.

Reply:
Done :)